app/medInria/medHomepageArea.cpp

```diff
@@ -443,10 +443,21 @@ void medHomepageArea::initPage()
     workspaceButtonsLayout->setColumnMinimumWidth(1,120);
     workspaceButtonsLayout->setColumnMinimumWidth(2,120);
     workspaceButtonsLayout->setColumnMinimumWidth(3,120);
-    workspaceButtonsLayout->addLayout(workspaceButtonsLayoutBasic,0,0);
-    workspaceButtonsLayout->addLayout(workspaceButtonsLayoutMethodology,0,1);
-    workspaceButtonsLayout->addLayout(workspaceButtonsLayoutClinical,0,2);
-    workspaceButtonsLayout->addLayout(workspaceButtonsLayoutOther,0,3);
+
+    // Hide a category if empty
+    QList<QVBoxLayout*> layoutList;
+    layoutList << workspaceButtonsLayoutBasic << workspaceButtonsLayoutMethodology
+               << workspaceButtonsLayoutClinical << workspaceButtonsLayoutOther;
+    int columnCount = 0;
+    foreach (QVBoxLayout* curLayout, layoutList)
+    {
+        if (!curLayout->isEmpty())
+        {
+            workspaceButtonsLayout->addLayout(curLayout,0,columnCount);
+            columnCount++;
+        }
+    }
+
```

Comment:
This contains two bugs:
- The column layouts are initially created without parents, so if one of the columns isn't added in the end, the column (and its label) won't get deleted.
- ```isEmpty``` shouldn't be true if there are no buttons, because the column always contains the title label. After some extensive googling (QT doesn't document this sort of behaviour very well), it seems that ```isEmpty``` only takes into account non hidden objects, and since the label is created without a parent, it is hidden by default. If you do a ``` workspaceButtonsLayoutOther->setParent(this)``` further up, you'll see that the column won't be hidden anymore. You should therefore not be relying on ```isEmpty``` to test if a column should be included or not.